bin/main.rb

```diff
@@ -1,2 +1,32 @@
-#!/usr/bin/main ruby
-puts 'Hello World'
+#!/usr/bin/env ruby
+def create_player
+  puts 'Please enter name of the first player'
+  player1 = gets.chomp
+  puts "Welcome #{player1}"
+  puts 'Please enter name of the second player'
+  player2 = gets.chomp
+  puts "Welcome #{player2}"
+end
+create_player
+
+def player_key
+  puts ' Choose your key'
+  key1 = gets.chomp
+  puts "Thanks for choosing #{key1}"
+  puts ' Choose your key'
+  key2 = gets.chomp
+  puts "Thanks for choosing #{key2}"
+end
+player_key
+
+def board_display
+  puts 'Here is the Board'
+  board = Array.new(9, 0)
+  i = 0
+  while i < board.length
+    print "#{board[i]} |" if i % 3 != 2
+    puts "#{board[i]} " if i % 3 == 2
+    i += 1
+  end
+end
+board_display
```

Comment:
You are missing flow for the logic. 
The board supposed to be displayed for each turn and be ended by some logical condition.
Also, you ought to mock player turn behavior and state whether a win or draw move was made.

SEe this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png) from odin

Reply:
It is recommended you use a while loop for the looping